custom/src/main/java/co/elastic/otel/config/DynamicInstrumentation.java

```diff
@@ -0,0 +1,261 @@
+                    getInstrumentationScopeInfo(sdkTracer);
+                TracerConfig tConfig = getTracerConfig(finalProvider, instrumentationScopeInfo);
+                Field tracerEnabledField = sdkTracer.getClass().getDeclaredField("tracerEnabled");
+                tracerEnabledField.setAccessible(true);
+                // Update is synced but the reader is NOT necessarily so this is eventual
+                // consistency, takes effect when the application passes a sync boundary
+                synchronized (sdkTracer) {
+                  tracerEnabledField.set(sdkTracer, tConfig.isEnabled());
+                }
+                if (logger.isLoggable(Level.CONFIG)) {
+                  String name = instrumentationScopeInfo.getName();
+                  if (name.startsWith(INSTRUMENTATION_NAME_PREPEND)) {
+                    name = name.substring(INSTRUMENTATION_NAME_PREPEND.length());
+                  }
+                  activatedTracers.add(name);
+                  activatedTracers.add(tConfig.isEnabled() ? "enabled" : "disabled");
+                }
+              } catch (NoSuchFieldException | IllegalAccessException e) {
+                throw new RuntimeException(e);
+              }
+            });
+    if (logger.isLoggable(Level.CONFIG)) {
+      logger.log(Level.CONFIG, "Activated Tracers: " + activatedTracers);
+    }
+  }
+
+  public static void reenableTracesFor(String instrumentationName) {
+    UpdatableConfigurator.INSTANCE.put(
+        InstrumentationScopeInfo.create(INSTRUMENTATION_NAME_PREPEND + instrumentationName),
+        TracerConfig.enabled());
+    updateTracerConfigurations(GlobalOpenTelemetry.getTracerProvider());
+  }
+
+  public static void disableTracesFor(String instrumentationName) {
+    UpdatableConfigurator.INSTANCE.put(
+        InstrumentationScopeInfo.create(INSTRUMENTATION_NAME_PREPEND + instrumentationName),
+        TracerConfig.disabled());
+    updateTracerConfigurations(GlobalOpenTelemetry.getTracerProvider());
+  }
+
+  public static class UpdatableConfigurator implements ScopeConfigurator<TracerConfig> {
+    public static final UpdatableConfigurator INSTANCE = new UpdatableConfigurator();
+    private final ConcurrentMap<String, TracerConfig> map = new ConcurrentHashMap<>();
+
+    private UpdatableConfigurator() {}
+
+    @Override
+    public TracerConfig apply(InstrumentationScopeInfo scopeInfo) {
+      return map.getOrDefault(scopeInfo.getName(), TracerConfig.defaultConfig());
+    }
+
+    public void put(InstrumentationScopeInfo scope, TracerConfig tracerConfig) {
+      map.put(scope.getName(), tracerConfig);
+    }
+  }
+
+  static {
+    if ("true".equals(System.getProperty(INSTRUMENTATION_DISABLE_OPTION + ".checker"))
+        || "true".equals(System.getenv("ELASTIC_OTEL_JAVA_DISABLE_INSTRUMENTATIONS_CHECKER"))) {
+      new Thread(new OptionChecker(), "Elastic dynamic_instrumentation checker").start();
```

Comment:
This will start a non daemon thread, which is a bad Idea IMO.

I'd suggest to do the following:
* Move the checker start logic into `DynamicInstrumentation.setTracerConfigurator`
* Add the `ConfigProperties` as argument to that method: This means you don't have to check env variables / proeprties and to the conversion yourself
* There, create a single-threaded daemon executor service with your periodic checker
* For a proper shutdown, I've used a trick before: On the `TracerProviderBuilder`, register a SpanProcessor which does nothing. It will still receive a shutdown event though, which can be used to properly shutdown the executor

Reply:
> - Move the checker start logic into DynamicInstrumentation.setTracerConfigurator

Although that works now, it just shifts needing to spawn it in a static initializer to a later implementation because setTracerConfigurator is really just a proxy for `SdkTracerProviderBuilder.setTracerConfigurator` until that latter becomes public, at which point `DynamicInstrumentation.setTracerConfigurator` disappears entirely - as do most of the DynamicInstrumentation methods.

I think it's fine to just make it a daemon thread - good catch! done that now

Reply:
Good for me, we can have another look later when things get more stable/public